src/main/java/org/openrewrite/staticanalysis/ReplaceLambdaWithMethodReference.java

```diff
@@ -54,16 +54,29 @@ public Duration getEstimatedEffortPerOccurrence() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-        return new JavaVisitor<ExecutionContext>() {
+        return new TreeVisitor<Tree, ExecutionContext>() {
             @Override
+            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx, Cursor parent) {
+                if (tree instanceof J.CompilationUnit) {
+                    return new ReplaceLambdaWithMethodReferenceJavaVisitor().visit(tree, ctx);
+                } else if (tree instanceof K.CompilationUnit) {
+                    return new ReplaceLambdaWithMethodReferenceKotlinVisitor().visit(tree, ctx);
+                }
+                return tree;
+            }
+        };
+    }
+
+    private static class ReplaceLambdaWithMethodReferenceKotlinVisitor extends KotlinVisitor<ExecutionContext> {
+        // Implement Me
+    }
```

Comment:
What's the plan here? Should this be logged as an issue for improvement @kunli2 ?
Or is it mostly to prevent exceptions and not make any changes just yet?

Reply:
created https://github.com/openrewrite/rewrite-static-analysis/issues/138 for tracking.